hazelcast/include/hazelcast/client/MixedMap.h

```diff
@@ -0,0 +1,1134 @@
+            * otherwise it will slow down the system.
+            *
+            * Warning 2: Do not make a call to hazelcast. It can cause deadlock.
+            *
+            * @param listener     entry listener
+            * @param predicate The query filter to use when returning the events to the user.
+            * @param includeValue <tt>true</tt> if <tt>EntryEvent</tt> should
+            *                     contain the value.
+            *
+            * @return registrationId of added listener that can be used to remove the entry listener.
+            */
+            std::string
+            addEntryListener(MixedEntryListener &listener, const query::Predicate &predicate, bool includeValue);
+
+            /**
+            * Removes the specified entry listener
+            * Returns silently if there is no such listener added before.
+            *
+            *
+            * @param registrationId id of registered listener
+            *
+            * @return true if registration is removed, false otherwise
+            */
+            bool removeEntryListener(const std::string &registrationId);
+
+            /**
+            * Adds the specified entry listener for the specified key.
+            *
+            * Warning 1: If listener should do a time consuming operation, off-load the operation to another thread.
+            * otherwise it will slow down the system.
+            *
+            * Warning 2: Do not make a call to hazelcast. It can cause deadlock.
+            *
+            * @param listener     entry listener
+            * @param key          key to listen
+            * @param includeValue <tt>true</tt> if <tt>EntryEvent</tt> should
+            *                     contain the value.
+            */
+            template <typename K>
+            std::string addEntryListener(MixedEntryListener &listener, const K &key, bool includeValue) {
+                serialization::pimpl::Data keyData = toData(key);
+                client::impl::MixedEntryEventHandler<protocol::codec::MapAddEntryListenerCodec::AbstractEventHandler> *entryEventHandler =
+                        new client::impl::MixedEntryEventHandler<protocol::codec::MapAddEntryListenerCodec::AbstractEventHandler>(
+                                getName(), context->getClusterService(), context->getSerializationService(),
+                                listener,
+                                includeValue);
+                return proxy::IMapImpl::addEntryListener(entryEventHandler, keyData, includeValue);
+            }
+
+            /**
+            * Returns the <tt>EntryView</tt> for the specified key.
+            *
+            *
+            * @param key key of the entry
+            * @return <tt>EntryView</tt> of the specified key
+            * @see EntryView
+            */
+            EntryView <TypedData, TypedData> getEntryView(const TypedData &key);
```

Comment:
Why key is TypedData instead of directly key itself ?

Reply:
Corrected.